ios/PurchasesHybridCommon/PurchasesHybridCommon/PaywallProxy.swift

```diff
@@ -36,46 +41,28 @@ import UIKit
         return controller
     }
 
-    @available(*, deprecated, message: "Use presentPaywall with paywallResultHandler instead")
-    @objc
-    public func presentPaywall() {
-        self.privatePresentPaywall(displayCloseButton: nil, offering: nil)
-    }
-
-    @available(*, deprecated, message: "Use presentPaywall with paywallResultHandler instead")
-    @objc
-    public func presentPaywall(displayCloseButton: Bool) {
-        self.privatePresentPaywall(displayCloseButton: displayCloseButton, offering: nil)
-    }
-
-    @available(*, deprecated, message: "Use presentPaywall with paywallResultHandler instead")
     @objc
-    public func presentPaywall(offering: Offering) {
-        self.privatePresentPaywall(displayCloseButton: nil, offering: offering)
-    }
+    public func createFooterPaywallView(offeringIdentifier: String) -> UIViewController {
+        let controller = PaywallFooterViewController(offeringIdentifier: offeringIdentifier)
+        controller.delegate = self
 
-    @available(*, deprecated, message: "Use presentPaywall with paywallResultHandler instead")
-    @objc
-    public func presentPaywall(offering: Offering, displayCloseButton: Bool) {
-        self.privatePresentPaywall(displayCloseButton: displayCloseButton, offering: offering)
+        return controller
     }
 
     @objc
     public func presentPaywall(paywallResultHandler: @escaping (String) -> Void) {
-        self.privatePresentPaywall(displayCloseButton: nil, offering: nil, paywallResultHandler: paywallResultHandler)
+        self.privatePresentPaywall(paywallResultHandler: paywallResultHandler)
     }
 
     @objc
     public func presentPaywall(displayCloseButton: Bool, paywallResultHandler: @escaping (String) -> Void) {
         self.privatePresentPaywall(displayCloseButton: displayCloseButton,
-                                   offering: nil,
                                    paywallResultHandler: paywallResultHandler)
     }
 
     @objc
     public func presentPaywall(offering: Offering, paywallResultHandler: @escaping (String) -> Void) {
```

Comment:
There's really no reason we have all these overloads. Better to have just a couple of methods and define default values for things like "displayCloseButton" in the hybrids, instead of checking 100 different permutations of parameters to decide which method to call.

Reply:
I agree, it makes adding parameters very complicated. I thought about creating an options object instead so we can have just that one parameter and defer the defaults to the natives

Reply:
Right... I think the idea was to try to avoid duplicating the defaults in the hybrids and rely on the ones in the natives... But yeah, it's likely not worth that 👍

Reply:
actually, that way we guarantee both platforms default to the same, because right now the default in iOS for `displayCloseButton` is false and for Android true